compiler/rustc_trait_selection/src/traits/select/candidate_assembly.rs

```diff
@@ -628,7 +628,10 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
         }
 
         self.infcx.probe(|_snapshot| {
-            if obligation.has_non_region_late_bound() {
+            // FIXME(non_lifetime_binders): Really, we care only to check that we don't
+            // have any non-region *escaping* bound vars, but that's hard to check and
```

Comment:
Can't we just augment `HasEscapingVarsVisitor` with the bound var kind we are looking for and create `TypeVisitableExt::has_escaping_non_lifetime_bound_vars` ?

Reply:
I'm not certain how that would work given the implementation of the visitor, since outer_exclusive_binder returns a debruijn index and does no additional walking

Reply:
Hmm, I guess you are right. It's more involved than I thought.